alertlogic/dynapi.py

```diff
@@ -188,14 +188,17 @@ def substitute_path_args(path, args):
 
 def substitute_query_args(query, args):
     parsed = urlparse.parse_qs(query)
+    qs_to_remove = {}
     for key in parsed:
         value = parsed[key][0]
         if value.startswith(":"):
             required_arg = value[1:]  # removes ":" at the beginning of the string
             if required_arg in args:
                 parsed[key] = args[required_arg]
             else:
-                raise InvalidEndpointCall("missing required url argument {}".format(required_arg))
+                qs_to_remove[key] = key
```

Comment:
**STRONG** cosmetic:  You are using dictionary as a list if i understand correctly. You need to do:
```python
qs_to_remove = []
...
qs_to_remove.append(key)
```